### ISSUE TITLE
Routes Grouping  with .post HookTrigger methods doesn't take into account group prefix

### ISSUE THREAD

**Issue by jaysonng** [netreconlab/parse-server-swift#54]
When grouping methods together with 

https://github.com/netreconlab/parse-server-swift/blob/0538fcaac76f508e524554b6a498c4e020fc9c98/Sources/ParseServerSwift/Extensions/ParseHookTrigger%2BVapor.swift#L891-L900

the prefix of the group, ie "user"
    
``` swift
    func boot(routes: RoutesBuilder) throws {
        
        let usersGroup = routes.grouped("user")

        usersGroup.post("login", "after",
                        object: User.self,
                        trigger: .afterLogin,
                        use: afterLogin
        )
    
    }
```
![Screenshot 2024-01-10 at 10 31 15 AM](https://github.com/netreconlab/parse-server-swift/assets/440445/943751b8-a677-49fb-9c2a-ea81d0d03966)

which produces a ParseError when called

```
  Error: {"message":{"status":404,"headers":{"content-length":"35","content-type":"application/json; charset=utf-8","connection":"keep-alive","date":"Wed, 10 Jan 2024 02:20:30 GMT"},"buffer":{"type":"Buffer","data":[123,34,114,101,97,115,111,110,34,58,34,78,111,116,32,70,111,117,110,100,34,44,34,101,114,114,111,114,34,58,116,114,117,101,125]},"text":"{\"reason\":\"Not Found\",\"error\":true}","data":{"reason":"Not Found","error":true}},"code":141} {"className":"_User","error":{"code":141,"message":{"buffer":{"data":[123,34,114,101,97,115,111,110,34,58,34,78,111,116,32,70,111,117,110,100,34,44,34,101,114,114,111,114,34,58,116,114,117,101,125],"type":"Buffer"},"data":{"error":true,"reason":"Not Found"},"headers":{"connection":"keep-alive","content-length":"35","content-type":"application/json; charset=utf-8","date":"Wed, 10 Jan 2024 02:20:30 GMT"},"status":404,"text":"{\"reason\":\"Not Found\",\"error\":true}"}},"triggerType":"afterLogin","user":"WnjQEA6FsE"}
/Users/jayson/Projects/heroku/node_modules/parse-server/lib/ParseServer.js:261
          throw err;
          ^

ParseError: [object Object]
    at resolveError (/Users/jayson/Projects/heroku/node_modules/parse-server/lib/triggers.js:657:17)
    at error (/Users/jayson/Projects/heroku/node_modules/parse-server/lib/triggers.js:431:17)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 141
}
```

the afterLogin method is the same from the routes.swift but modified to work with Collections

``` swift
func afterLogin(req: Request) async throws -> ParseHookResponse<Bool> {

        // Note that `ParseHookResponse<Bool>` means a "successful"
        // response will return a "Bool" type. Bool is the standard response with
        // a "true" response meaning everything is okay or continue.
        if let error: ParseHookResponse<Bool> = checkHeaders(req) {
            return error
        }
        let parseRequest = try req.content
            .decode(ParseHookTriggerObjectRequest<User, User>.self)
        
        req.logger.info("A user has logged in: \(parseRequest)")
        
        return ParseHookResponse(success: true)
    }
```

I tried pinpointing where it all goes wrong in [ParseHookTrigger%2BVapor.swift](https://github.com/netreconlab/parse-server-swift/blob/main/Sources/ParseServerSwift/Extensions/ParseHookTrigger%2BVapor.swift) but couldn't. 

to make it work, I had to make the code as such:

```swift

    func boot(routes: RoutesBuilder) throws {
        
        //let usersGroup = routes.grouped("user")

        routes.post("user", "login", "after",
                        object: User.self,
                        trigger: .afterLogin,
                        use: afterLogin
        )
                
    }
    
```

thanks,

**Comment by cbaker6** [comment]
Can you explain how this differs from https://github.com/netreconlab/parse-server-swift/discussions/38#discussioncomment-6255285 and not fixed in #41?

**Comment by jaysonng** [comment]
This only happened now when I first tried using a HookTrigger. 

The last time they I was using it for HookFunctions and was fixed by your update.

so it has something to do with the methods that use the .post(_ path: PathComponent..., object: trigger: use:) I believe.

**Comment by jaysonng** [comment]
Hmm. reminding me of #41 led me to check the changes you made on ParseHookFunction.swift and I think copying that change to ParseHookTrigger.swift fixed the issue.

I'll do a pull request.

**Comment by cbaker6** [comment]
Solved in #55
